src/modules/endpoint/sagas/watchFetchCategory.js

```diff
@@ -9,32 +9,45 @@ import type {
 } from '../../app/StoreActionType'
 import type { DataContainer } from '../DataContainer'
 import loadCityContent from './loadCityContent'
+import { ContentLoadCriterion } from '../ContentLoadCriterion'
+import DatabaseContext from '../DatabaseContext'
+import { LanguageModel } from '@integreat-app/integreat-api-client'
 
 function * fetchCategory (dataContainer: DataContainer, action: FetchCategoryActionType): Saga<void> {
-  const {city, language, path, depth, key, forceUpdate, shouldRefreshResources} = action.params
+  const { city, language, path, depth, key, criterion } = action.params
   try {
-    yield call(loadCityContent, dataContainer, city, language, forceUpdate, shouldRefreshResources)
+    const loadCriterion = new ContentLoadCriterion(criterion)
+    const languageLoaded = yield call(loadCityContent, dataContainer, city, language, loadCriterion)
 
-    const [categoriesMap, resourceCache, languages] = yield all([
-      call(dataContainer.getCategoriesMap),
-      call(dataContainer.getResourceCache),
-      call(dataContainer.getLanguages)
-    ])
+    if (languageLoaded) {
+      const context = new DatabaseContext(city, language)
+      const [categoriesMap, resourceCache] = yield all([
+        call(dataContainer.getCategoriesMap, context),
+        call(dataContainer.getResourceCache, context)
+      ])
 
-    const insert: PushCategoryActionType = {
-      type: `PUSH_CATEGORY`,
-      params: {
-        categoriesMap,
-        resourceCache,
-        path,
-        languages,
-        depth,
-        key,
-        city,
-        language
+      let languages = [new LanguageModel(language, language)]
+
+      if (!loadCriterion.peek()) {
+        languages = yield call(dataContainer.getLanguages, context)
```

Comment:
Why only load the languages if we aren't peeking?

Reply:
If we are peeking there is no need for the languages as the language switch is disabled.

The languages of the opened city will be in the state and not those of the peeked city.